File: openc3-cosmos-init/plugins/packages/openc3-cosmos-demo/targets/INST/procedures/stash.rb
# Stash API is useful for storing simple key/value pairs
# to preserve state between script runs
stash_set('key1', 'val1')
stash_set('key2', 'val2')
check_expression("'#{stash_get('key1')}' == 'val1'")
check_expression("'#{stash_get('key2')}' == 'val2'")
check_expression("'#{stash_keys().to_s}' == '[\"key1\", \"key2\"]'")
stash_set('key1', 1)
stash_set('key2', 2)
check_expression("'#{stash_all().to_s}' == '{\"key1\"=>1, \"key2\"=>2}'")
stash_delete('key2')
check_expression("#{stash_get('key2').nil?} == true")
stash_delete('key1')
data = [1,2,[3,4]]
stash_set('ary', data)
check_expression("'#{stash_get('ary')}' == '#{data.to_s}'")
stash_delete('ary')
hash = { one: 1, two: 2, string: 'string' }
stash_set('hash', hash)
check_expression("'#{stash_get('hash')}' == '#{hash.to_s}'")
stash_delete('hash')
os = OpenStruct.new
os.company = "OpenC3"
os.software = "COSMOS"
stash_set('os', os)
check_expression("'#{stash_get('os').to_s}' == '#{os.to_s}'")
stash_delete('os')

File: openc3-cosmos-init/plugins/packages/openc3-cosmos-demo/targets/INST2/procedures/stash.py
# Stash API is useful for storing simple key/value pairs
# to preserve state between script runs
stash_set("key1_py", "val1")
stash_set("key2_py", "val2")
check_expression(f"'{stash_get('key1_py')}' == 'val1'")
check_expression(f"'{stash_get('key2_py')}' == 'val2'")
stash_set("key1_py", 1)
stash_set("key2_py", 2)
check_expression(f"'{stash_get('key1_py')}' == '1'")
check_expression(f"'{stash_get('key2_py')}' == '2'")
stash_delete("key2_py")
check_expression(f"{stash_get('key2_py')} == None")
stash_delete("key1_py")
data = [1, 2, [3, 4]]
stash_set("ary_py", data)
check_expression(f"{stash_get('ary_py')} == {data}")
stash_delete("ary_py")
data = {"one": 1, "two": 2, "string": "string"}
stash_set("data_py", data)
check_expression(f"{stash_get('data_py')} == {data}")
stash_delete("data_py")
